// react-image-upload/dist/index.js
function _interopDefault (ex) { return (ex && (typeof ex === 'object') && 'default' in ex) ? ex['default'] : ex; }

var React = require('react');
var React__default = _interopDefault(React);

function _extends() {
  _extends = Object.assign || function (target) {
    for (var i = 1; i < arguments.length; i++) {
      var source = arguments[i];

      for (var key in source) {
        if (Object.prototype.hasOwnProperty.call(source, key)) {
          target[key] = source[key];
        }
      }
    }

    return target;
  };

  return _extends.apply(this, arguments);
}

var styles = {"__content":"_2VJLH","__imgWrapper":"_FW764","__btnWrapper":"_354pS","__btn":"_3TA7V","__label":"_3ulBn","__input":"_1Zcq6","__img":"_196dV"};

var ImageUploader = function ImageUploader(_ref) {
  var onFileAdded = _ref.onFileAdded,
      onFileRemoved = _ref.onFileRemoved,
      uploadElement = _ref.uploadElement,
      height = _ref.height,
      width = _ref.width,
      background = _ref.background;
  var ref = React.useRef();

  var _useState = React.useState({}),
      currentImg = _useState[0],
      setCurrentImg = _useState[1];

  var handleImageUpload = function handleImageUpload(e) {
    if (e.target.files.length > 0) {
      var imageObject = {
        file: e.target.files[0],
        dataUrl: URL.createObjectURL(e.target.files[0])
      };
      setCurrentImg(function (oldImage) {
        return _extends({}, oldImage, imageObject);
      });
      if (onFileAdded) onFileAdded(currentImg);
    }
  };

  var handleDeleteImage = function handleDeleteImage() {
    if (onFileRemoved) onFileRemoved(currentImg);
    ref.current.files = null;
    ref.current.value = '';
    setCurrentImg({});
  };

  return /*#__PURE__*/React__default.createElement("div", {
    className: styles.__content
  }, /*#__PURE__*/React__default.createElement("div", {
    className: styles.__imgWrapper,
    style: {
      height: height,
      width: width,
      background: background
    }
  }, /*#__PURE__*/React__default.createElement("div", {
    className: styles.__btnWrapper
  }, /*#__PURE__*/React__default.createElement("button", {
    className: styles.__btn,
    onClick: handleDeleteImage
  }, /*#__PURE__*/React__default.createElement("svg", {
    xmlns: "http://www.w3.org/2000/svg",
    width: "20",
    height: "20",
    viewBox: "0 0 24 24",
    fill: "#848484"
  }, /*#__PURE__*/React__default.createElement("path", {
    d: "M12 0c-6.627 0-12 5.373-12 12s5.373 12 12 12 12-5.373 12-12-5.373-12-12-12zm7 14h-14v-4h14v4z"
  })))), /*#__PURE__*/React__default.createElement("label", {
    className: styles.__label
  }, /*#__PURE__*/React__default.createElement(UploadElement, {
    element: uploadElement
  }), currentImg.dataUrl ? /*#__PURE__*/React__default.createElement(React__default.Fragment, null) : /*#__PURE__*/React__default.createElement("input", {
    key: currentImg,
    ref: ref,
    type: "file",
    name: "upload",
    className: styles.__input,
    onChange: handleImageUpload,
    accept: "image/*",
    id: "upload"
  })), currentImg.dataUrl ? /*#__PURE__*/React__default.createElement("img", {
    className: styles.__img,
    src: currentImg === null || currentImg === void 0 ? void 0 : currentImg.dataUrl,
    alt: currentImg === null || currentImg === void 0 ? void 0 : currentImg.dataUrl,
    loading: "lazy"
  }) : /*#__PURE__*/React__default.createElement(React__default.Fragment, null)));
};

function UploadElement(_ref2) {
  var element = _ref2.element;

  if (element) {
    return /*#__PURE__*/React__default.createElement(React__default.Fragment, null, element);
  }

  return /*#__PURE__*/React__default.createElement("svg", {
    width: "50",
    height: "50",
    viewBox: "0 0 50 50",
    fill: "none",
    xmlns: "http://www.w3.org/2000/svg"
  }, /*#__PURE__*/React__default.createElement("path", {
    d: "M8.33332 8.33332H14.5833L18.75 4.16666H31.25L35.4167 8.33332H41.6667C42.7717 8.33332 43.8315 8.77231 44.6129 9.55371C45.3943 10.3351 45.8333 11.3949 45.8333 12.5V37.5C45.8333 38.6051 45.3943 39.6649 44.6129 40.4463C43.8315 41.2277 42.7717 41.6666 41.6667 41.6666H8.33332C7.22825 41.6666 6.16845 41.2277 5.38704 40.4463C4.60564 39.6649 4.16666 38.6051 4.16666 37.5V12.5C4.16666 11.3949 4.60564 10.3351 5.38704 9.55371C6.16845 8.77231 7.22825 8.33332 8.33332 8.33332ZM25 14.5833C22.2373 14.5833 19.5878 15.6808 17.6343 17.6343C15.6808 19.5878 14.5833 22.2373 14.5833 25C14.5833 27.7627 15.6808 30.4122 17.6343 32.3657C19.5878 34.3192 22.2373 35.4167 25 35.4167C27.7627 35.4167 30.4122 34.3192 32.3657 32.3657C34.3192 30.4122 35.4167 27.7627 35.4167 25C35.4167 22.2373 34.3192 19.5878 32.3657 17.6343C30.4122 15.6808 27.7627 14.5833 25 14.5833ZM25 18.75C26.6576 18.75 28.2473 19.4085 29.4194 20.5806C30.5915 21.7527 31.25 23.3424 31.25 25C31.25 26.6576 30.5915 28.2473 29.4194 29.4194C28.2473 30.5915 26.6576 31.25 25 31.25C23.3424 31.25 21.7527 30.5915 20.5806 29.4194C19.4085 28.2473 18.75 26.6576 18.75 25C18.75 23.3424 19.4085 21.7527 20.5806 20.5806C21.7527 19.4085 23.3424 18.75 25 18.75Z",
    fill: "#848484"
  }));
}

module.exports = ImageUploader;
